spec/services/record_participant_declaration_service_spec.rb

```diff
@@ -5,85 +5,119 @@
 RSpec.describe RecordParticipantDeclaration do
   let(:ecf_lead_provider) { create(:lead_provider) }
   let(:cpd_lead_provider) { create(:cpd_lead_provider, lead_provider: ecf_lead_provider) }
-  let(:another_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
+  let(:another_cpd_lead_provider) { create(:cpd_lead_provider, name: "Unknown") }
```

Comment:
I smell... conflicts.

In here and in shared setup, with the work I've done on https://github.com/DFE-Digital/early-careers-framework/pull/859/files#diff-b04de0b996bbd7c765305b53215f91f9edad6ef4df41f5fe83615d74a3ffb898L5

I'm happy to deal with the shared setup conflicts myself, but could you use the diff from the link above to make this test use common track and pay test setup? When you do, I hope things will 'just work' - and if not I am happy to pair on it.

Reply:
I think it's just easier to fix the conflicts after the merge. Feel free to merge first and I'll patch up afterwards

Reply:
oh wait. i misunderstood and was reading another part of the PR. hmm do we really want these as shared contexts? i foresee these being painful sharing all this setup stuff. if someone wants to test something slightly different it just won't work as it's fully coupled. I'd rather opt for duplication here personally.

i see shared contexts more of a way to test for a very specific behaviour common to many things. are we even in that area?

Reply:
I think we want a base shared setup for T&P tests - and to diverge from that setup if some individual tests require it. I think almost all T&P tests require some ECF participant in a school which is in a partnership with a lead provider, and writing it down everywhere is a pain. This is simply a place where we haven't used the shared setup yet (it is a pretty new thing).